packages/typegpu/src/types.ts

```diff
@@ -14,6 +14,25 @@ export type Resource = {
   dataType: AnyTgpuData | UnknownData;
 };
 
+export type TgpuShaderStage = 'compute' | 'vertex' | 'fragment';
+
+/**
+ * Removes properties from record type that extend `Prop`
+ */
+export type OmitProps<T extends Record<string, unknown>, Prop> = Pick<
+  T,
+  {
+    [Key in keyof T]: T[Key] extends Prop ? never : Key;
+  }[keyof T]
+>;
+
+/**
+ * Utility type that takes an object type and makes the hover overlay more readable.
+ */
+export type Prettify<T> = {
```

Comment:
what difference does it make? and where do we plan on using it? right now it's unused

Reply:
It was being used before, but no longer is. I will remove it for now.